components/omega/src/ocn/OceanInit.cpp

```diff
@@ -0,0 +1,217 @@
+
+   // extract variables for time management group
+   Config TimeMgmtConfig("TimeManagement");
+   if (OmegaConfig->existsGroup("TimeManagement")) {
+      int Err1 = OmegaConfig->get(TimeMgmtConfig);
+      // check requested calendar is a valid option, return error if not found
+      if (TimeMgmtConfig.existsVar("CalendarType")) {
+         std::string ConfigCalStr;
+         CalendarKind ConfigCalKind = CalendarUnknown;
+         I4 Err1     = TimeMgmtConfig.get("CalendarType", ConfigCalStr);
+         I4 ICalType = 9;
+         for (I4 I = 0; I < NUM_SUPPORTED_CALENDARS; ++I) {
+            if (ConfigCalStr == CalendarKindName[I]) {
+               ICalType      = I;
+               ConfigCalKind = (CalendarKind)(ICalType + 1);
+               break;
+            }
+         }
+         if (ICalType == 9) {
+            LOG_ERROR("ocnInit: Requested Calendar type not found");
+            ++RetErr;
+            return RetErr;
+         }
+         // destroy default Calendar to keep static NumCalendars member
+         // accurate, then construct requested Calendar
+         OmegaCal.~Calendar();
+         OmegaCal = Calendar(ConfigCalStr, ConfigCalKind);
+      }
+
+      // check for start time and set if found
+      if (TimeMgmtConfig.existsVar("StartTime")) {
+         std::string StartTimeStr;
+         I4 Err1 = TimeMgmtConfig.get("StartTime", StartTimeStr);
+
+         StartTime = TimeInstant(&OmegaCal, StartTimeStr);
+      }
+
+      std::string NoneStr("none");
+      // set RunInterval by checking for StopTime and RunDuration in Config,
+      // if both are present and inconsistent, use RunDuration
+      if (TimeMgmtConfig.existsVar("StopTime")) {
+         std::string StopTimeStr;
+         I4 Err1 = TimeMgmtConfig.get("StopTime", StopTimeStr);
+         if (StopTimeStr != NoneStr) {
+            TimeInstant StopTime(&OmegaCal, StopTimeStr);
+            RunInterval = StopTime - StartTime;
+         }
+      }
+      if (TimeMgmtConfig.existsVar("RunDuration")) {
+         std::string RunDurationStr;
+         I4 Err1 = TimeMgmtConfig.get("RunDuration", RunDurationStr);
+         if (RunDurationStr != NoneStr) {
+            TimeInterval IntervalFromStr(RunDurationStr);
+            if (IntervalFromStr != RunInterval) {
+               LOG_WARN("ocnInit: RunDuration and StopTime are inconsistent: "
+                        "using RunDuration");
+               RunInterval = IntervalFromStr;
+            }
+         }
+      }
+   }
+
+   // return error if RunInterval set to zero
+   if (RunInterval == ZeroInterval) {
+      LOG_ERROR("ocnInit: Simulation run duration set to zero");
+      ++RetErr;
+   }
+
+   // set EndAlarm based on length of RunInterval
+   TimeInstant EndTime = StartTime + RunInterval;
+   EndAlarm            = Alarm("End Alarm", EndTime);
+
+   return RetErr;
+} // end initTimeManagement
+
+// Call init routines for remaining Omega modules
+int initOmegaModules(MPI_Comm Comm) {
```

Comment:
This could be used to for initialization in the test drivers to reduce repeat code, but maybe its better to not use it since it would initialize modules that aren't actually needed for a given test.